src/DeviceInterfaceValues.tsx

```diff
@@ -184,6 +191,260 @@ const InterfaceData = ({ interfaceData }: InterfaceDataProps): React.ReactElemen
   );
 };
 
+interface SendInterfaceDataModalProps {
+  showModal: boolean;
+  interfaceDefinition: AstarteInterface;
+  sendingData: boolean;
+  handleShowModal: () => void;
+  sendInterfaceData: (data: { endpoint: string; value: any }) => void;
+}
+
+const SendInterfaceDataModal = ({
+  showModal,
+  sendingData,
+  interfaceDefinition,
+  handleShowModal,
+  sendInterfaceData,
+}: SendInterfaceDataModalProps) => {
+  const [selectedMapping, setSelectedMapping] = useState<AstarteMapping | null>(null);
+  const [endpoint, setEndpoint] = useState('');
+  const [endpointWithParams, setEndpointWithParams] = useState('');
+  const [paramValues, setParamValues] = useState<{ [key: string]: string }>({});
+  const [pathParams, setPathParams] = useState<string[]>([]);
+  const [value, setValue] = useState('');
+  const [parsedIndividualValue, setParsedIndividualValue] = useState<AstarteDataValue>();
+  const [data, setData] = useState<{ [key: string]: string }>({});
+  const [parsedObjectData, setParsedObjectData] = useState<{ [key: string]: AstarteDataValue }>({});
+  const [errors, setErrors] = useState<{ [key: string]: string }>({});
+
+  const parseValue = (type: AstarteDataType, value: string) => {
+    switch (type) {
+      case 'integer':
+      case 'longinteger':
+        return parseInt(value, 10);
+      case 'double':
+        return parseFloat(value);
+      case 'boolean':
+        return value.toLowerCase() === 'true';
+      case 'doublearray':
+      case 'integerarray':
+      case 'booleanarray':
+      case 'longintegerarray':
+      case 'stringarray':
+      case 'binaryblobarray':
+      case 'datetimearray':
+        return JSON.parse(value);
+      default:
+        return value;
```

Comment:
if we match all the different `AstarteDataType` there is no need for a `default:` case here